src/synapse/azext_synapse/_validators.py

```diff
@@ -4,18 +4,26 @@
 # --------------------------------------------------------------------------------------------
 
 
-def example_name_or_id_validator(cmd, namespace):
-    #  pylint: disable=line-too-long
-    # Example of a storage account name or ID validator.
-    # See: https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-name-or-id-parameters
-    from azure.cli.core.commands.client_factory import get_subscription_id
-    from msrestazure.tools import is_valid_resource_id, resource_id
-    if namespace.storage_account:
-        if not is_valid_resource_id(namespace.RESOURCE):
-            namespace.storage_account = resource_id(
-                subscription=get_subscription_id(cmd.cli_ctx),
-                resource_group=namespace.resource_group_name,
-                namespace='Microsoft.Storage',
-                type='storageAccounts',
-                name=namespace.storage_account
-            )
+from msrestazure.tools import is_valid_resource_id
+from azure.cli.core.commands.parameters import get_resources_in_subscription  # pylint: disable=unused-import
+
+
+def validate_storage_account(namespace):
+    from msrestazure.tools import parse_resource_id
+    if is_valid_resource_id(namespace.storage_account):
+        parsed_workspace = parse_resource_id(namespace.storage_account)
```

Comment:
>parsed_workspace [](start = 8, length = 16)

parsed_storage

Reply:
fixed

---
In reply to: [424957674](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r424957674) [](ancestors = 424957674)